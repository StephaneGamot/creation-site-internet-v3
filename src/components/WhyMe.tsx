"use client";

import { motion } from 'framer-motion';
import { useTranslations } from 'next-intl';

type WhyMeItem = {
  title: string;
  description: string;
};

export default function WhyMe() {
  const t = useTranslations('whyMe');
  const whyMeList = t.raw('list') as WhyMeItem[];

  return (
    <section
      role="region"
      aria-labelledby="whyme-title"
      className="max-w-6xl py-10 mx-auto my-8"
    >
      <h2
        id="whyme-title"
        className="text-3xl font-title font-semibold mb-8 text-center"
      >
        {t('title')}
      </h2>

      <div className="grid gap-8 md:grid-cols-3">
        {whyMeList.map((item, index) => (
          <motion.div
            key={index}
            tabIndex={0}
            initial={{ opacity: 0, y: 20 }}
            whileInView={{ opacity: 1, y: 0 }}
            transition={{ delay: index * 0.2 }}
            className="block border border-gray-200 hover:border-[#3451a1] bg-white hover:bg-[#3451a1]/5 p-6 rounded-2xl shadow-sm hover:shadow-md transition-all duration-300"
          >
            <h3 className="text-xl text-center text-[#0d0d0d] font-title font-semibold mb-2">
              {item.title}
            </h3>
            <p className="text-sm text-gray-600 text-left">{item.description}</p>
          </motion.div>
        ))}
      </div>
    </section>
  );
}
