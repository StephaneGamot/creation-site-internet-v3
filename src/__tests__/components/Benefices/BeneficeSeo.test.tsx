import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { axe } from 'jest-axe';
import WhyMe from '@/components/WhyMe';

// ✅ Mock de Framer Motion (pour éviter les erreurs liées à l’animation dans Jest)
jest.mock('framer-motion', () => ({
  motion: {
    div: ({ children, ...rest }: any) => <div {...rest}>{children}</div>,
  },
}));

// ✅ Mock de next-intl
jest.mock('next-intl', () => ({
  useTranslations: () => {
    const translations = {
      title: 'Pourquoi me choisir ?',
      list: [
        {
          title: 'Expertise Full-Stack',
          description: 'Une maîtrise complète du cycle de développement web.',
        },
        {
          title: 'SEO & Performance',
          description: 'Des sites pensés pour Google et ultra rapides.',
        },
        {
          title: 'Design Élégant',
          description: 'Un sens aigu de l’esthétique et de l’expérience utilisateur.',
        },
      ],
    };

    const t = (key: keyof typeof translations) =>
      typeof translations[key] === 'string'
        ? (translations[key] as string)
        : key;

    t.raw = (key: keyof typeof translations) => translations[key];
    return t;
  },
}));

describe('WhyMe component', () => {
  it('affiche le titre traduit', () => {
    render(<WhyMe />);
    expect(
      screen.getByRole('heading', { name: /pourquoi me choisir/i })
    ).toBeInTheDocument();
  });

  it('affiche tous les éléments de la liste', () => {
    render(<WhyMe />);
    expect(screen.getByText('Expertise Full-Stack')).toBeInTheDocument();
    expect(screen.getByText('SEO & Performance')).toBeInTheDocument();
    expect(screen.getByText('Design Élégant')).toBeInTheDocument();
  });

  it('passe les tests d’accessibilité', async () => {
    const { container } = render(<WhyMe />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
