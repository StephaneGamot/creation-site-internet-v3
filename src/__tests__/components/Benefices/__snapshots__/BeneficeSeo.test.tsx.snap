// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BeneficeSeo correspond au snapshot 1`] = `null`;
